src/main/java/uk/gov/di/ipv/cri/passport/lambda/PassportHandler.java

```diff
@@ -131,12 +120,84 @@ public APIGatewayProxyResponseEvent handleRequest(
+            AuthenticationRequest.parse(queryStringParameters);
+        } catch (ParseException e) {
+            LOGGER.error(("Failed to parse oath query string parameters: " + e.getMessage()));
+            throw new HttpResponseExceptionWithErrorBody(
+                    HttpStatus.SC_BAD_REQUEST,
+                    ErrorResponse.FAILED_TO_PARSE_OAUTH_QUERY_STRING_PARAMETERS);
+        }
+    }
+
+    private PassportFormRequest parsePassportFormRequest(String input)
+            throws HttpResponseExceptionWithErrorBody {
+        LOGGER.info("Parsing passport form data into payload for DCS");
+        try {
+            return objectMapper.readValue(input, PassportFormRequest.class);
+        } catch (JsonProcessingException e) {
+            LOGGER.error(("Failed to parse payload from input: " + e.getMessage()));
+            throw new HttpResponseExceptionWithErrorBody(
+                    HttpStatus.SC_BAD_REQUEST, ErrorResponse.FAILED_TO_PARSE_PASSPORT_FORM_DATA);
+        }
+    }
 
-        return ValidationResult.createValidResult();
+    private JWSObject preparePayload(PassportFormRequest passportFormRequest)
+            throws HttpResponseExceptionWithErrorBody {
+        LOGGER.info("Preparing payload for DCS");
+        try {
+            return dcsCryptographyService.preparePayload(passportFormRequest);
+        } catch (CertificateException
+                | NoSuchAlgorithmException
+                | InvalidKeySpecException
+                | JOSEException
+                | JsonProcessingException e) {
+            LOGGER.error(("Failed to prepare payload for DCS: " + e.getMessage()));
+            throw new HttpResponseExceptionWithErrorBody(
+                    HttpStatus.SC_INTERNAL_SERVER_ERROR,
+                    ErrorResponse.FAILED_TO_PREPARE_DCS_PAYLOAD);
+        }
+    }
+
+    private DcsSignedEncryptedResponse doPassportCheck(JWSObject preparedPayload)
+            throws HttpResponseExceptionWithErrorBody {
+        LOGGER.info("Sending passport check to DCS");
+        try {
+            return passportService.dcsPassportCheck(preparedPayload);
+        } catch (IOException e) {
+            LOGGER.error(("Passport check with DCS failed: " + e.getMessage()));
+            throw new HttpResponseExceptionWithErrorBody(
+                    HttpStatus.SC_INTERNAL_SERVER_ERROR, ErrorResponse.ERROR_CONTACTING_DCS);
+        }
+    }
+
+    private DcsResponse unwrapDcsResponse(DcsSignedEncryptedResponse response)
+            throws HttpResponseExceptionWithErrorBody {
+        LOGGER.info("Unwrapping DCS response");
+        try {
+            return dcsCryptographyService.unwrapDcsResponse(response);
+        } catch (CertificateException
+                | java.text.ParseException
+                | JOSEException
+                | JsonProcessingException e) {
+            LOGGER.error(("Failed to unwrap response from DCS: " + e.getMessage()));
+            throw new HttpResponseExceptionWithErrorBody(
```

Comment:
I'm enjoying this pattern of logging the error then throwing in all these helper functions.